### PR TITLE
VST3 better native UI size handling 

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #3233

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/effects/VST3/internal/PlugFrame.cpp [hunk 1]
```diff
@@ -23,26 +24,32 @@ internal::PlugFrame::~PlugFrame()
...
 {
    if(auto window = mWindow.get())
    {
-      auto fixed = view->canResize() != Steinberg::kResultTrue;
-      UpdateWindowSize(window, {newSize->getWidth(), newSize->getHeight()}, fixed);
+      auto topWindow = wxGetTopLevelParent(window);
+
+      wxWindowUpdateLocker windowUpdateLocker(topWindow);
+
+      window->SetInitialSize({newSize->getWidth(), newSize->getHeight() });
```
Comment (Paul-Licameli) [inline src/effects/VST3/internal/PlugFrame.cpp:41]: A deleted comment said
```
           //No need to accommodate the off-by-one error with Steinberg::ViewRect
           //as we do it with wxRect
```

Are you sure there are no implications of off-by-one errors here in the rewrite?

**Review by Paul-Licameli — APPROVED** [review]
I am not expert in the details of X11 UI code but I have a general understanding that certain settings of window sizes need to happen earlier.  No serious objections, but one suggestion for clarifying the use of bitwise operations.

File: src/effects/VST3/VST3Effect.cpp [hunk 2]
```diff
@@ -236,14 +236,11 @@ bool VST3Effect::LoadFactoryPreset(int id, EffectSettings& settings) const
...
+   EffectUIValidator *validator, bool forceModal)
 {
-   if(!IsGraphicalUI())
-   {
-      //Restrict resize of the "plain" dialog
-      dialog.SetMaxSize(dialog.GetSize());
-      dialog.SetMinSize(dialog.GetSize());
-   }
+   if(validator->IsGraphicalUI())
+      dialog.SetWindowStyle(dialog.GetWindowStyle() ^ (wxRESIZE_BORDER | wxMAXIMIZE_BOX));
```
[the remark this answers is not in the document]
Comment (vsverchinsky) [reply src/effects/VST3/VST3Effect.cpp:242]: Yes, it's not a correct expression, thanks!

File: src/effects/VST3/internal/PlugFrame.cpp (hunk 1 above)
Comment (vsverchinsky) [reply src/effects/VST3/internal/PlugFrame.cpp:41]: Where is that error?

**Pull request merged**